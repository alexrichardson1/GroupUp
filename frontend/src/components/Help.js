import { Component } from "react";
import React from "react";
import { Button, Card } from "react-bootstrap";
import ReactCardFlip from "react-card-flip";
import Navbar from "components/NavBar";

export default class Help extends Component {
  constructor() {
    super();
    this.state = {
      flipped: new Set(),
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(id) {
    return (e) => {
      e.preventDefault();
      let flipped = new Set(this.state.flipped);
      flipped.has(id) ? flipped.delete(id) : flipped.add(id);
      this.setState({ flipped });
    };
  }

  componentDidMount() {
    document.title = "Need Help?";
  }

  render() {
    return (
      <div>
        <Navbar renderBool={[false, false, false, false]} helpPage={true} />

        <ReactCardFlip
          isFlipped={this.state.flipped.has(1)}
          flipDirection="horizontal"
        >
          <Card>
            <Card.Body>
              <Card.Title>What Is A Hackathon?</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/hackathons.webp"}
              />
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(1)}>Click to flip</Button>
            </Card.Footer>
          </Card>

          <Card>
            <Card.Body>
              <Card.Title>Information</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/hackathons.webp"}
              />
              <Card.Text>
                Hackathons are a time limited event where teams work together to
                build software for an array of reasons. They are a great way of
                sharpening your design technique, refining your programming
                ability and making friends. Use GroupUp to find your dream
                hackathon team! Use our filters to narrow down groups based on
                the programming language used, their time zone and more!
              </Card.Text>
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(1)}>Click to flip</Button>
            </Card.Footer>
          </Card>
        </ReactCardFlip>

        <ReactCardFlip
          isFlipped={this.state.flipped.has(2)}
          flipDirection="horizontal"
        >
          <Card>
            <Card.Body>
              <Card.Title>How To Join A Group</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/groups.webp"}
              />
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(2)}>Click to flip</Button>
            </Card.Footer>
          </Card>

          <Card>
            <Card.Body>
              <Card.Title>Information</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/groups.webp"}
              />
              <Card.Text>
                Choose a Hackathon. Groups should be displayed for your chosen
                Hackathon. Make use of the filters if you have specific
                requirements. Click 'Find out more', then click 'Join Group'
              </Card.Text>
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(2)}>Click to flip</Button>
            </Card.Footer>
          </Card>
        </ReactCardFlip>

        <ReactCardFlip
          isFlipped={this.state.flipped.has(3)}
          flipDirection="horizontal"
        >
          <Card>
            <Card.Body>
              <Card.Title>How To Advertise Your Group</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/advertise.jpg"}
              />
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(3)}>Click to flip</Button>
            </Card.Footer>
          </Card>

          <Card>
            <Card.Body>
              <Card.Title>Information</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/advertise.jpg"}
              />
              <Card.Text>
                Choose a Hackathon. Click 'Advertise my group!'. Fill out the
                required information. Click 'Submit'.
              </Card.Text>
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(3)}>Click to flip</Button>
            </Card.Footer>
          </Card>
        </ReactCardFlip>

        <ReactCardFlip
          isFlipped={this.state.flipped.has(4)}
          flipDirection="horizontal"
        >
          <Card>
            <Card.Body>
              <Card.Title>How do hackathons work?</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/hackathons2.jpg"}
              />
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(4)}>Click to flip</Button>
            </Card.Footer>
          </Card>

          <Card>
            <Card.Body>
              <Card.Title>Information</Card.Title>
              <Card.Img
                variant="top"
                className="cardPicture"
                src={process.env.PUBLIC_URL + "/hackathons2.jpg"}
              />
              <Card.Text>
                A theme is given to you and a time limit. Work together with
                your team to produce some software that is linked to the theme
                in that time limit and present it to try and win the best
                project award!
              </Card.Text>
            </Card.Body>
            <Card.Footer>
              <Button onClick={this.handleClick(4)}>Click to flip</Button>
            </Card.Footer>
          </Card>
        </ReactCardFlip>
      </div>
    );
  }
}
