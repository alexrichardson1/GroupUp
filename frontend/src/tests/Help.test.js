import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Help from "components/Help";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Help />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
