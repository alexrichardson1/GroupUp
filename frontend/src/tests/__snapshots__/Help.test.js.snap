// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navBar navbar navbar-expand-lg navbar-light fixed-top"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "color": "white",
        }
      }
    >
      <img
        alt="group up logo"
        className="logoImage"
        href="/home"
        onClick={[Function]}
        src="https://i.imgur.com/w92t1xH.png"
      />
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr=auto navbar-nav"
        onKeyDown={[Function]}
      >
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
           
        </span>
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Help
        </span>
      </div>
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="ml-auto navHelp navbar-brand"
          href="/help"
          id="collasible-nav-dropdown"
          onClick={[Function]}
          title="Help"
        >
          <img
            alt="help"
            className="splitterImage"
            src="/info-circle.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <div
    className="react-card-flip"
    style={
      Object {
        "perspective": "1000px",
        "zIndex": "auto",
      }
    }
  >
    <div
      className="react-card-flipper"
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className="react-card-front"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "relative",
            "top": "0",
            "transform": "rotateY(0deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
            "zIndex": "2",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              What Is A Hackathon?
            </div>
            <img
              className="card-img-top cardPicture"
              src="/hackathons.webp"
            />
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
      <div
        className="react-card-back"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "absolute",
            "top": "0",
            "transform": "rotateY(-180deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Information
            </div>
            <img
              className="card-img-top cardPicture"
              src="/hackathons.webp"
            />
            <p
              className="card-text"
            >
              Hackathons are a time limited event where teams work together to build software for an array of reasons. They are a great way of sharpening your design technique, refining your programming ability and making friends. Use GroupUp to find your dream hackathon team! Use our filters to narrow down groups based on the programming language used, their time zone and more!
            </p>
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="react-card-flip"
    style={
      Object {
        "perspective": "1000px",
        "zIndex": "auto",
      }
    }
  >
    <div
      className="react-card-flipper"
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className="react-card-front"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "relative",
            "top": "0",
            "transform": "rotateY(0deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
            "zIndex": "2",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              How To Join A Group
            </div>
            <img
              className="card-img-top cardPicture"
              src="/groups.webp"
            />
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
      <div
        className="react-card-back"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "absolute",
            "top": "0",
            "transform": "rotateY(-180deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Information
            </div>
            <img
              className="card-img-top cardPicture"
              src="/groups.webp"
            />
            <p
              className="card-text"
            >
              Choose a Hackathon. Groups should be displayed for your chosen Hackathon. Make use of the filters if you have specific requirements. Click 'Find out more', then click 'Join Group'
            </p>
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="react-card-flip"
    style={
      Object {
        "perspective": "1000px",
        "zIndex": "auto",
      }
    }
  >
    <div
      className="react-card-flipper"
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className="react-card-front"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "relative",
            "top": "0",
            "transform": "rotateY(0deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
            "zIndex": "2",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              How To Advertise Your Group
            </div>
            <img
              className="card-img-top cardPicture"
              src="/advertise.jpg"
            />
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
      <div
        className="react-card-back"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "absolute",
            "top": "0",
            "transform": "rotateY(-180deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Information
            </div>
            <img
              className="card-img-top cardPicture"
              src="/advertise.jpg"
            />
            <p
              className="card-text"
            >
              Choose a Hackathon. Click 'Advertise my group!'. Fill out the required information. Click 'Submit'.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="react-card-flip"
    style={
      Object {
        "perspective": "1000px",
        "zIndex": "auto",
      }
    }
  >
    <div
      className="react-card-flipper"
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className="react-card-front"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "relative",
            "top": "0",
            "transform": "rotateY(0deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
            "zIndex": "2",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              How do hackathons work?
            </div>
            <img
              className="card-img-top cardPicture"
              src="/hackathons2.jpg"
            />
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
      <div
        className="react-card-back"
        style={
          Object {
            "WebkitBackfaceVisibility": "hidden",
            "backfaceVisibility": "hidden",
            "height": "100%",
            "left": "0",
            "position": "absolute",
            "top": "0",
            "transform": "rotateY(-180deg)",
            "transformStyle": "preserve-3d",
            "transition": "0.6s",
            "width": "100%",
          }
        }
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Information
            </div>
            <img
              className="card-img-top cardPicture"
              src="/hackathons2.jpg"
            />
            <p
              className="card-text"
            >
              A theme is given to you and a time limit. Work together with your team to produce some software that is linked to the theme in that time limit and present it to try and win the best project award!
            </p>
          </div>
          <div
            className="card-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Click to flip
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
